diff --git a/configure b/configure
index 4286c34..b4301fa 100644
--- a/configure
+++ b/configure
@@ -1297,7 +1297,7 @@ fi

 [ -n "$relaxed_plugins" ] && DEFS="$DEFS -DRELAXED_PLUGINS"

-CFLAGS_GEN=" -std=gnu99 -Wall -Wextra"
+CFLAGS_GEN=" -std=gnu99 -Wall -Wextra -fcommon"
 if test -z "$debug" ; then
 	CFLAGS="$CFLAGS $CFLAGS_GEN -fvisibility=hidden"

diff --git a/src/mk_server.c b/src/mk_server.c
index 8c27742..956f7ee 100644
--- a/src/mk_server.c
+++ b/src/mk_server.c
@@ -40,11 +40,9 @@
 unsigned int mk_server_worker_capacity(unsigned short nworkers)
 {
     unsigned int max, avl;
-    struct rlimit lim;
-
-    /* Limit by system */
-    getrlimit(RLIMIT_NOFILE, &lim);
-    max = lim.rlim_cur;
+
+    /* Overwrite the value for RLIMIT_NOFILE to avoid unmanagable memory allocation */
+    max = 1048576;

     /* Minimum of fds needed by Monkey:
      * --------------------------------

